### PR TITLE
avoid div by zero in extreme balance case

### PR DESCRIPTION
**Pull request by djrtwo**
Address #1663

* `get_total_balance` returns a minimum of `EFFECTIVE_BALANCE_INCREMENT` to avoid div by zero in extreme cases
* add rewards test vector to test for this extreme case (it did in fact div by 0 before the fix was in place)
* Add another rewards test for empty attestations (because I noticed it was needed... )

### CODE REVIEW

File: specs/phase0/beacon-chain.md [hunk 1]
```diff
@@ -996,10 +996,11 @@ def get_beacon_proposer_index(state: BeaconState) -> ValidatorIndex:
 ```python
 def get_total_balance(state: BeaconState, indices: Set[ValidatorIndex]) -> Gwei:
     """
-    Return the combined effective balance of the ``indices``. (1 Gwei minimum to avoid divisions by zero.)
+    Return the combined effective balance of the ``indices``.
+    ``EFFECTIVE_BALANCE_INCREMENT`` Gwei minimum to avoid divisions by zero.
```
Comment (hwwhww) [inline specs/phase0/beacon-chain.md]: Consider also noting in `get_total_active_balance`,`get_attesting_balance` comments?

IMHO the change adds some code debts, but still tolerable for the edge case. :)

**Comment by djrtwo** [comment]
Thanks @hwwhww!
Made the suggested changes. Merging after tests run
